plugins/module_utils/openshift_builds.py

```diff
@@ -0,0 +1,638 @@
+                namespace = build["metadata"]["namespace"]
+                resource.delete(name=name, namespace=namespace, body={})
+            except DynamicApiError as exc:
+                msg = "Failed to delete Build %s/%s due to: %s" % (namespace, name, exc.body)
+                self.fail_json(msg=msg, status=exc.status, reason=exc.reason)
+            except Exception as e:
+                msg = "Failed to delete Build %s/%s due to: %s" % (namespace, name, to_native(e))
+                self.fail_json(msg=msg, error=to_native(e), exception=e)
+        self.exit_json(changed=changed, builds=candidates)
+
+    def clone_build(self, name, namespace, request):
+        try:
+            result = self.client.request(
+                method="POST",
+                path="/apis/build.openshift.io/v1/namespaces/{namespace}/builds/{name}/clone".format(
+                    namespace=namespace,
+                    name=name
+                ),
+                body=request,
+                content_type="application/json",
+            )
+            return result.to_dict()
+        except DynamicApiError as exc:
+            msg = "Failed to clone Build %s/%s due to: %s" % (namespace, name, exc.body)
+            self.fail_json(msg=msg, status=exc.status, reason=exc.reason)
+        except Exception as e:
+            msg = "Failed to clone Build %s/%s due to: %s" % (namespace, name, to_native(e))
+            self.fail_json(msg=msg, error=to_native(e), exception=e)
+
+    def instantiate_build_config(self, name, namespace, request):
+        try:
+            result = self.client.request(
+                method="POST",
+                path="/apis/build.openshift.io/v1/namespaces/{namespace}/buildconfigs/{name}/instantiate".format(
+                    namespace=namespace,
+                    name=name
+                ),
+                body=request,
+                content_type="application/json",
+            )
+            return result.to_dict()
+        except DynamicApiError as exc:
+            msg = "Failed to instantiate BuildConfig %s/%s due to: %s" % (namespace, name, exc.body)
+            self.fail_json(msg=msg, status=exc.status, reason=exc.reason)
+        except Exception as e:
+            msg = "Failed to instantiate BuildConfig %s/%s due to: %s" % (namespace, name, to_native(e))
+            self.fail_json(msg=msg, error=to_native(e), exception=e)
+
+    def git_ref_info(self, path, ref):
+        path = self.params.get("git-repository")
+        result = {}
+        if path:
+            if not HAS_GIT_MODULE:
+                self.module.fail_json(
+                    msg="The git module should be installed. Try `pip install gitpython`",
+                    exception=GIT_MODULE_ERROR,
+                    error=to_native(git_import_exception),
+                )
+            repo = git.Repo(path)
+            git = repo.git
+            ret = git.show("-s", ref, '--format="%H%n%an%n%ae%n%cn%n%ce%n%B"')
```

Comment:
This seems to be the only place we're using the gitpython dependency. All this appears to do is shell out to the git binary. Can we just do that ourselves and drop the python dependency?

Reply:
will do this, we will change the requirement from gitpython to ``git`` binary